ftplugin/tex.lua

```diff
@@ -1,4 +1,64 @@
+local preview_settings = {}
+
+local sumatrapdf_args = { "-reuse-instance", "%p", "-forward-search", "%f", "%l" }
+local evince_args = { "-f", "%l", "%p", "\"code -g %f:%l\"" }
+local okular_args = { "--unique", "file:%p#src:%l%f" }
+local zathura_args = { "--synctex-forward", "%l:1:%f", "%p" }
+local qpdfview_args = { "--unique", "%p#src:%f:%l:1" }
+local skim_args = { "%l", "%p", "%f" }
+
+if O.lang.latex.forward_search.executable == "C:/Users/{User}/AppData/Local/SumatraPDF/SumatraPDF.exe" then
+  preview_settings = sumatrapdf_args
+elseif O.lang.latex.forward_search.executable == "evince-synctex" then
+  preview_settings = evince_args
+elseif O.lang.latex.forward_search.executable == "okular" then
+  preview_settings = okular_args
+elseif O.lang.latex.forward_search.executable == "zathura" then
+  preview_settings = zathura_args
+elseif O.lang.latex.forward_search.executable == "qpdfview" then
+  preview_settings = qpdfview_args
+elseif O.lang.latex.forward_search.executable == "/Applications/Skim.app/Contents/SharedSupport/displayline" then
+  preview_settings = skim_args
+elseif O.lang.latex.forward_search.executable == "" then
+  preview_settings = {}
+end
+
 require("lspconfig").texlab.setup {
   cmd = { DATA_PATH .. "/lspinstall/latex/texlab" },
   on_attach = require("lsp").common_on_attach,
+  handlers = {
+    ["textDocument/publishDiagnostics"] = vim.lsp.with(vim.lsp.diagnostic.on_publish_diagnostics, {
+      virtual_text = O.lang.latex.diagnostics.virtual_text,
+      signs = O.lang.latex.diagnostics.signs,
+      underline = O.lang.latex.diagnostics.underline,
+       update_in_insert = true,
+    }),
+  },
+  filetypes = { "tex", "bib" },
+  settings = {
+    texlab = {
+      auxDirectory = O.lang.latex.aux_directory,
+      bibtexFormatter = O.lang.latex.bibtex_formatter,
+      build = {
+        args = { '-pdf', '-interaction=nonstopmode', '-synctex=1', '%f' },
```

Comment:
I tested passing the configuration entry and can guarantee you that it does work.
Please update it

Reply:
Ok. You are right. For a moment I thought it works because it falls back to the defaults. But no.
This set me off:
![build_args_are_empty](https://user-images.githubusercontent.com/86895284/124570420-26353e80-de47-11eb-8cea-4562344a4b92.png)

I glad that it works with the configuration through lv-config.lua
And done. :)

Reply:
Yep I know, that was misleading for me too :)